src/Aspire.Hosting/ApplicationModel/ContainerPullPolicyAnnotation.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Image pull policies for container resources.
+/// </summary>
+public enum PullPolicy
+{
+    /// <summary>
+    /// Always pull the image when creating the container.
+    /// </summary>
+    Always,
+    /// <summary>
+    /// Pull the image only if it does not already exist.
+    /// </summary>
+    Missing,
+    /// <summary>
+    /// Never pull the image.
+    /// </summary>
+    Never
```

Comment:
Why would we want `Never`?

Reply:
It's definitely an edge case setting; it's supported in DCP, but I'm happy to leave it out of the Aspire types until such time as we get feedback asking for it.